grails-app/assets/javascripts/esp-overview.js

```diff
@@ -127,6 +127,16 @@ function selectReportingPeriod(project) {
     var currentReport;
     if (selectedYear) {
         currentReport = findReportFromFinancialYear(project.reports,selectedYear);
+
+        // will fetch the latest report with filters
```

Comment:
This block should be moved outside of the "else" of the if (selectedYear) so it applies to both methods of report selection.
Also, best to create the "now" date outside of the loop.